tests/testthat/helper-integration-tests-setup.R

```diff
@@ -0,0 +1,186 @@
+  fishing_fleet$estimate_q <- FALSE
+  fishing_fleet$random_q <- FALSE
+  fishing_fleet$log_obs_error <- rep(log(sqrt(log(em_input$cv.L$fleet1^2 + 1))), om_input$nyr)
+  fishing_fleet$estimate_obs_error <- FALSE
+  # Modules are linked together using module IDs
+  # Each module has a get_id() function that returns the unique ID for that module
+  # Each fleet uses the module IDs to link up the correct module to the correct fleet
+  # Note: Likelihoods not yet set up as a stand-alone modules, so no get_id()
+  fishing_fleet$SetAgeCompLikelihood(1)
+  fishing_fleet$SetIndexLikelihood(1)
+  fishing_fleet$SetSelectivity(fishing_fleet_selectivity$get_id())
+  fishing_fleet$SetObservedIndexData(fishing_fleet_index$get_id())
+  fishing_fleet$SetObservedAgeCompData(fishing_fleet_age_comp$get_id())
+
+  # Create the survey fleet
+  survey_fleet_selectivity <- new(LogisticSelectivity)
+  survey_fleet_selectivity$inflection_point$value <- om_input$sel_survey$survey1$A50.sel1
+  survey_fleet_selectivity$inflection_point$is_random_effect <- FALSE
+  # turn on estimation of inflection_point
+  survey_fleet_selectivity$inflection_point$estimated <- TRUE
+  survey_fleet_selectivity$slope$value <- om_input$sel_survey$survey1$slope.sel1
+  survey_fleet_selectivity$slope$is_random_effect <- FALSE
+  # turn on estimation of slope
+  survey_fleet_selectivity$slope$estimated <- TRUE
+
+  survey_fleet <- new(Fleet)
+  survey_fleet$is_survey <- TRUE
+  survey_fleet$nages <- om_input$nages
+  survey_fleet$nyears <- om_input$nyr
+  survey_fleet$estimate_F <- FALSE
+  survey_fleet$random_F <- FALSE
+  survey_fleet$log_q <- log(om_output$survey_q$survey1)
+  survey_fleet$estimate_q <- TRUE
+  survey_fleet$random_q <- FALSE
+  survey_fleet$log_obs_error <- rep(log(sqrt(log(em_input$cv.survey$survey1^2 + 1))), om_input$nyr)
+  survey_fleet$estimate_obs_error <- FALSE
+  survey_fleet$SetAgeCompLikelihood(1)
+  survey_fleet$SetIndexLikelihood(1)
+  survey_fleet$SetSelectivity(survey_fleet_selectivity$get_id())
+  survey_fleet$SetObservedIndexData(survey_fleet_index$get_id())
+  survey_fleet$SetObservedAgeCompData(survey_fleet_age_comp$get_id())
+
+  # Population
+  population <- new(Population)
+  population$log_M <- rep(log(om_input$M.age[1]), om_input$nyr * om_input$nages)
+  population$estimate_M <- FALSE
+  population$log_init_naa <- log(om_output$N.age[1, ])
+  population$estimate_init_naa <- TRUE
+  population$nages <- om_input$nages
+  population$ages <- om_input$ages
+  population$nfleets <- sum(om_input$fleet_num, om_input$survey_num)
+  population$nseasons <- 1
+  population$nyears <- om_input$nyr
+  population$SetMaturity(maturity$get_id())
+  population$SetGrowth(ewaa_growth$get_id())
+  population$SetRecruitment(recruitment$get_id())
+
+  # Set-up TMB
+  CreateTMBModel()
+  # Create parameter list from Rcpp modules
+  parameters <- list(p = get_fixed())
+  obj <- TMB::MakeADFun(data = list(), parameters, DLL = "FIMS", silent=TRUE)
```

Comment:
I added `silent=TRUE` here to declutter the console. It prevents printing the mgc at each iteration.